src/rt/sections_elf_shared.d

```diff
@@ -894,42 +965,58 @@ struct tls_index
     size_t ti_offset;
 }
 
-extern(C) void* __tls_get_addr(tls_index* ti) nothrow @nogc;
+version (OSX)
+{
+    extern(C) void _d_dyld_getTLSRange(void*, void**, size_t*) nothrow @nogc;
 
-/* The dynamic thread vector (DTV) pointers may point 0x8000 past the start of
- * each TLS block. This is at least true for PowerPC and Mips platforms.
- * See: https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/powerpc/dl-tls.h;h=f7cf6f96ebfb505abfd2f02be0ad0e833107c0cd;hb=HEAD#l34
- *      https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/mips/dl-tls.h;h=93a6dc050cb144b9f68b96fb3199c60f5b1fcd18;hb=HEAD#l32
- */
-version (X86)
-    enum TLS_DTV_OFFSET = 0x0;
-else version (X86_64)
-    enum TLS_DTV_OFFSET = 0x0;
-else version (ARM)
-    enum TLS_DTV_OFFSET = 0x0;
-else version (AArch64)
-    enum TLS_DTV_OFFSET = 0x0;
-else version (SPARC)
-    enum TLS_DTV_OFFSET = 0x0;
-else version (SPARC64)
-    enum TLS_DTV_OFFSET = 0x0;
-else version (PPC)
-    enum TLS_DTV_OFFSET = 0x8000;
-else version (PPC64)
-    enum TLS_DTV_OFFSET = 0x8000;
-else version (MIPS32)
-    enum TLS_DTV_OFFSET = 0x8000;
-else version (MIPS64)
-    enum TLS_DTV_OFFSET = 0x8000;
+    void[] getTLSRange(void *tlsSymbol) nothrow @nogc
+    {
+        void* start = null;
+        size_t size = 0;
+        _d_dyld_getTLSRange(tlsSymbol, &start, &size);
+        assert(start && size, "Could not determine TLS range.");
```

Comment:
This check will be removed if asserts are removed, is that intentional?

Reply:
Most likely not => #2324.